docs/src/world_age_issue.md

```diff
@@ -0,0 +1,109 @@
+don't want to load any thing related to image IO.
+
+To avoid such unnecessary loading latency, FileIO defers package loading until it's actually used.
+For instance, when you use FileIO, you'll probably observe something like this:
+
+```julia
+julia> using TestImages, FileIO
+
+julia> path = testimage("cameraman"; download_only=true)
+"/home/jc/.julia/artifacts/27a4c26bcdd47eb717bee089ec231a899cb8ef69/cameraman.tif"
+
+julia> load(path) # actual backend loading happens here
+[ Info: Precompiling ImageIO [82e4d734-157c-48bb-816b-45c225c6df19]
+[ Info: Precompiling TiffImages [731e570b-9d59-4bfa-96dc-6df516fadf69]
+...
+```
+
+## The hidden issue
+
+Although this lazy loading trick solves the large loading latency issue, it isn't a normal practice
+in Julia because it introduces a so-called _world age issue_ or _world age problem_. The world age
+issue happens when the compiled function becomes outdated.
+
+Now, let's try something new to uncover this issue:
+
+```julia
+julia> using IndirectArrays, ImageCore
+
+julia> img = IndirectArray(rand(1:5, 4, 4), rand(RGB, 5))
+4×4 IndirectArray{RGB{Float64}, 2, Int64, Matrix{Int64}, Vector{RGB{Float64}}}:
+...
+
+julia> save("indexed_image.png", img)
+```
+
+and **reopen a new julia REPL** and wrap the `load` into a function, you'll observe something like
+this:
+
+```julia
+julia> using FileIO
+
+julia> f() = length(load("indexed_image.png"))
+f (generic function with 1 method)
+
+julia> f()
+ERROR: MethodError: no method matching size(::IndirectArrays.IndirectArray{ColorTypes.RGB{FixedPointNumbers.N0f8}, 2, UInt8, Matrix{UInt8}, OffsetArrays.OffsetVector{ColorTypes.RGB{FixedPointNumbers.N0f8}, Vector{ColorTypes.RGB{FixedPointNumbers.N0f8}}}})
+The applicable method may be too new: running in world age 32711, while current world is 32746.
+Closest candidates are:
+  size(::IndirectArrays.IndirectArray) at ~/.julia/packages/IndirectArrays/BUQO3/src/IndirectArrays.jl:52 (method too new to be called from this world context.)
+  size(::AbstractArray{T, N}, ::Any) where {T, N} at abstractarray.jl:42
+  size(::Union{LinearAlgebra.Adjoint{T, var"#s962"}, LinearAlgebra.Transpose{T, var"#s962"}} where {T, var"#s962"<:(AbstractVector)}) at ~/packages/julias/julia-latest/share/julia/stdlib/v1.9/LinearAlgebra/src/adjtrans.jl:173
+  ...
+Stacktrace:
+ [1] length(t::IndirectArrays.IndirectArray{ColorTypes.RGB{FixedPointNumbers.N0f8}, 2, UInt8, Matrix{UInt8}, OffsetArrays.OffsetVector{ColorTypes.RGB{FixedPointNumbers.N0f8}, Vector{ColorTypes.RGB{FixedPointNumbers.N0f8}}}})
+   @ Base ./abstractarray.jl:291
+```
+
+This is because when you first call function `f()`, the IndirectArrays package is not yet loaded and
+thus `size(::IndirectArray)` method is not defined yet. This is exactly what world age means: the
+compiled `f()` function is outdated because new methods are dynamically added in runtime. The
+perhaps surprising fact is that if you retry `f()`, it just works:
```

Comment:
```suggestion
To understand why this happened, you have to understand the order of events:

1. When calling `f()` from the REPL, Julia first compiled `f`. Importantly, when compiling, Julia didn't know what type of object was going to be returned by `load`, so in the compiled code it waits to see what object actually gets returned before figuring out which method of `size` to call. (This is called *runtime dispatch*.)
2. It queried the file, recognized a PNG file, and *loaded the ImageIO and PNGFiles packages*. (It's for the loading of these packages that you needed to start a fresh Julia session.)
3. FileIO calls the appropriate PNG-specific `load` function in PNGFiles. (We'll have more to say about this step further below.) This causes an image to be returned, which is an array of a type defined by the IndirectArrays package (a dependency of PNGFiles). 
4. `f` calls `size` on the returned image. However, this fails, because at the time you called `f`, the IndirectArrays package wasn't loaded.

In other words, `size` method lives in a world that's newer than the one from which you called `f()`.
This leads to the observed error.

!!! note
    World age is crucial to Julia's ability to allow
    you to *redefine* methods interactively, but the error we're illustrating is an unfortunate side-effect.

The good news is it's easy to fix, just try calling `f()` again:
```